### DISCUSSION TITLE
iCalendar-standard

### DISCUSSION THREAD

**Discussion by imperiumsources** [mhoeher/opentodolist#90] [General]
Hallo,

allgemein bin ich auf der Suche nach einen Tool dass sowohl auf meinen Desktop (GNOME4x) funktioniert als
auf meinen Android Smartphone [1]. Als App nutze ich das  jtx Board (https://jtx.techbee.at/)  das für sich den
iCalendar-Standard verwendet.  Das Tool ist soweit brauchbar. Synchronisieren tue ich es per Davx5. 

Ich habe keine Möglichkeit gefunden ihre Anwendung mit der Android App zu verbinden. 
Gibt es in Zukunft den CalDAV Support? 

[1] Die Android App Opentodolist ist nicht mein Geschmack.

**Comment by mhoeher** [comment]
Hallo @imperiumsources,

Um es kurz zu machen: CalDAV Support ist nichts, was zeitnah kommen wird. Die OpenTodoList hat bereits einige "Refactoring" Runden hinter sich - bei der letzten großen habe ich mich dediziert gegen CalDAV als "Backend" Format entschieden. Warum das? Weil es eher ein "Quasi Standard" ist.

Ja, viele Eigenschaften von Objekten sind festgelegt. Aber über Tools und Server hinweg gibt es nicht einmal einen einheitlichen Standard, um Hierarchien von Aufgaben (in OpenTodoList wird Todo List -> Todo -> Task verwendet) abzubilden. Das führt dann dazu, dass Hierarchien solcher Objekte zwar in App A funktionieren, wenn Benutzer die selbe Bibliothek/den selben Kalender dann allerdings mit App B verwenden wollen, kommt das böse Erwachen...

OpenTodoList verwendet daher ein eigenes Datenmodell und speichert alle Informationen als einfache Dateien in Verzeichnisstrukturen ab. Diese Verzeichnisse werden dann auf beliebige Server synchronisiert (aktuell ist WebDAV Support implementiert, mit spezieller Unterstützung für ownCloud und NextCloud, die beide WebDAV als Schnittstelle zur Synchronisation anbieten) sowie Dropbox. Support für beliebige weitere Server kann leicht nachgerüstet werden (da die App als einzige Anforderung hat, dass der Server das Speichern von Dateien in Verzeichnissen erlauben muss, so wie irgendeine Art von "Zeitstempel" bereithält, damit Änderungen auf dem Server erkannt werden können.

Theoretisch könnte man ein Backend anlegen, welches die Synchronisation mit einem CalDAV Server zulässt, allerdings dürfte das nicht trivial sein (die App unterstützt z.B. das Anhängen von Dateien an die meisten Einträge innerhalb einer Library - soweit ich weiß sind solche Feature in CalDAV nicht nativ enthalten, so dass auch hier nur selektiv einige Server als "Backend" unterstützt werden könnten).

Ich hoffe, dass beantwortet deine Frage erstmal!

Bzgl.:

>  [1] Die Android App Opentodolist ist nicht mein Geschmack.

Für Verbesserungsvorschläge bin ich immer offen - dazu entweder Issues/Tickets mit konkreten Vorschlägen öffnen, oder weitere Diskussionen hier starten. Ich muss allerdings auch dazu sagen, dass die OpenTodoList kein kommerzielles Tool ist und ich nur begrenzt viel Zeit in die App stecken kann. Ein dediziertes "Android UI" wird es daher nicht geben - das wäre zu viel Arbeit.

**Reply by imperiumsources** [reply]
Ich verstehe deine Beweggründe und die Daten als Webdav ist natürlich eine Möglichkeit zu speichern. Das Problem was ich
persönlich sehe ist diese Art als Insellösung und der Zugriff auf die Todos mit anderen Devices (z.b. Android Apps) ist nicht
möglich weil hier viele auf CalDav setzen. Vielleicht ist es aber auch gut so. So ist man flexibler. Man braucht nur ein Webdav und kann loslegen. Ich will das jetzt auch nicht vertiefen.  Deine App wurde so Designt Punkt.

Persönlich werde ich Deine Anwendung eine Zeitlang nutzen. Ich habe bereits ein Schwung an Ideen aber auch bei mir ist
Zeitlich eher knapp und  ich weis es aktuell nicht ob ich es hier im Forum oder als "Issues" reinschreibe. Aber das hat aktuell
noch bei mir Zeit.     

VG Richard

**Reply by mhoeher** [reply]
Vorweg: Verbesserungsvorschläge sind natürlich immer willkommen ;-) Gerne auch erstmal hier als Diskussionsbeitrag (falls möglich auch gern in Englisch - das erweitert die Runde derer, die dann evtl. auch mitlesen und vielleicht auch etwas schreiben).

Bzgl. der CalDAV Thematik: Ja, das ist die Wahl zwischen Pest und Cholera. Das Datenformat der OpenTodoList ist in der Tat eine Insellösung - aber eben auch eine, die unter allen Plattformen sauber funktioniert (vorausgesetzt, man kann mit OpenTodoList als Client auf allen Devices leben).

Für CalDAV gibt es ja diverse Anwendungen auf allen Systemen - aber leider sind diese dann doch nur begrenzt miteinander kompatibel. Ich wollte mir nicht den Stiefel anziehen und in diesem "Zoo" mitmischen.
